src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs

```diff
@@ -499,6 +499,23 @@ public T GetPayloadValue<T>(string key)
             return Payload.Value<T>(key);
         }
 
+        /// <summary>
+        /// Tries to get the 'value' corresponding to the provided key from the JWT payload { key, 'value' }.
+        /// </summary>
+        /// <remarks>If the key has no corresponding value, returns false. Otherwise returns true. </remarks>   
+        public bool TryGetPayloadValue<T>(string key, out T value)
+        {
+            if (string.IsNullOrEmpty(key))
+            {
+                value = default(T);
+                return false;
+            }
+
+            var result = Payload.TryGetValue(key, out var jTokenValue);
+            value = jTokenValue == null ? default(T) : jTokenValue.ToObject<T>();
```

Comment:
This will need to be in a try/catch as jTokenValue.ToObject<T> could throw.

Reply:
Fixed.